docs/apache-airflow/executor/debug.rst

```diff
@@ -15,11 +15,56 @@
     specific language governing permissions and limitations
     under the License.
 
+Testing DAGs with dag.test()
+=============================
+
+To debug DAGs in an IDE, you can set up the ``dag.test`` command in your dag file and run through your DAG in a single
+serialized python process.
+
+The ``dag.test`` command has the following benefits over the :class:`~airflow.executors.debug_executor.DebugExecutor`
+class, which is now deprecated:
+
+1. It does not require running an executor at all. Tasks are run one at a time with no executor or scheduler logs.
+2. It is significantly faster than running code with a DebugExecutor as it does not need to go through a scheduler loop
+3. It does not perform a backfill
+
+This executor can be used with any supported database (including a local SQLite database) and will
+``fail fast`` as all tasks run in a single process.
```

Comment:
I would move this to a subsection called `Comparison with DebugExecutor`

I think that what should be front and center, is just what it is / how to use it

Reply:
@dstandish done!